extension/parquet/parquet_reader.cpp

```diff
@@ -241,8 +243,14 @@ LogicalType ParquetReader::DeriveLogicalType(const SchemaElement &s_ele, bool bi
 			}
 		case ConvertedType::INTERVAL:
 			return LogicalType::INTERVAL;
-		case ConvertedType::JSON:
-			return LogicalType::VARCHAR;
+		case ConvertedType::JSON: {
+			if (context_ptr && Catalog::GetEntry<TypeCatalogEntry>(*context_ptr, INVALID_CATALOG, INVALID_SCHEMA,
```

Comment:
I would vote for always returning `LogicalType::JSON` and requiring the `JSON` extension to make use of Parquet files with JSON objects in them. We do the same with `TIMESTAMP_TZ` after all and will likely do the same when we add support for geoparquet.